src/docfx/lib/UrlUtility.cs

```diff
@@ -344,6 +344,68 @@ public static string RemoveLeadingHostName(string url, string hostName, bool rem
             : $"/{path}";
     }
 
+    public static string GetBookmark(string uid)
+    {
+        uid = uid.ToLowerInvariant();
+        var shouldIgnore = new HashSet<char>() { '\"', '\'', '%', '^', '\\' };
+        var shouldKeep = new HashSet<char>() { '(', ')', '*', '@' };
```

Comment:
This causes a new hashset allocation per `GetBookmark` call, this can be simplified by using `switch case` inside the for loop

Reply:
Removed.